tools/packaging/packaging.mk

```diff
@@ -1,8 +1,195 @@
 #remove leading characters since package version expects to start with digit
 PACKAGE_VERSION ?= $(shell echo $(VERSION) | sed 's/^[a-z]*-//' | sed 's/-//')
 
-# Building the debian file, docker.istio.deb and istio.deb
-include tools/packaging/deb/istio.mk
+# Creates the 2 packages. BUILD_WITH_CONTAINER=1 or in CI/CD (BUILD_WITH_CONTAINER=0)
+#
+# Development/manual testing:
+#    make deb          - builds debian packaging
+#    make deb/docker   - builds a test docker image
 
-# RPM/RHEL/CENTOS stuff
-include tools/packaging/rpm/rpm.mk
+deb: ${ISTIO_OUT_LINUX}/release/istio-sidecar.deb ${ISTIO_OUT_LINUX}/release/istio.deb
+
+# Base directory for istio binaries. Likely to change !
+ISTIO_DEB_BIN=/usr/local/bin
+
+ISTIO_DEB_DEPS:=pilot-discovery istioctl
+ISTIO_FILES:=
+$(foreach DEP,$(ISTIO_DEB_DEPS),\
+        $(eval ${ISTIO_OUT_LINUX}/release/istio.deb: $(ISTIO_OUT_LINUX)/$(DEP)) \
+        $(eval ISTIO_FILES+=$(ISTIO_OUT_LINUX)/$(DEP)=$(ISTIO_DEB_BIN)/$(DEP)) )
+
+SIDECAR_DEB_DEPS:=envoy pilot-agent
+SIDECAR_FILES:=
+$(foreach DEP,$(SIDECAR_DEB_DEPS),\
+        $(eval ${ISTIO_OUT_LINUX}/release/istio-sidecar.deb: $(ISTIO_OUT_LINUX)/$(DEP)) \
+        $(eval ${ISTIO_OUT_LINUX}/release/istio-sidecar.rpm: $(ISTIO_OUT_LINUX)/$(DEP)) \
+        $(eval SIDECAR_FILES+=$(ISTIO_OUT_LINUX)/$(DEP)=$(ISTIO_DEB_BIN)/$(DEP)) )
+
+ISTIO_DEB_DEST:=${ISTIO_DEB_BIN}/istio-start.sh \
+		/lib/systemd/system/istio.service \
+		/var/lib/istio/envoy/sidecar.env
+
+$(foreach DEST,$(ISTIO_DEB_DEST),\
+        $(eval ${ISTIO_OUT_LINUX}/istio-sidecar.deb:   tools/packaging/common/$(notdir $(DEST))) \
+        $(eval SIDECAR_FILES+=${REPO_ROOT}/tools/packaging/common/$(notdir $(DEST))=$(DEST)))
+
+SIDECAR_FILES+=${REPO_ROOT}/tools/packaging/common/envoy_bootstrap.json=/var/lib/istio/envoy/envoy_bootstrap_tmpl.json
+
+
+# original name used in 0.2 - will be updated to 'istio.deb' since it now includes all istio binaries.
+SIDECAR_PACKAGE_NAME ?= istio-sidecar
+
+# TODO: rename istio-sidecar.deb to istio.deb
+
+# Note: adding --deb-systemd ${REPO_ROOT}/tools/packaging/common/istio.service will result in
+# a /etc/systemd/system/multi-user.target.wants/istio.service and auto-start. Currently not used
+# since we need configuration.
+# --iteration 1 adds a "-1" suffix to the version that didn't exist before
+${ISTIO_OUT_LINUX}/release/istio-sidecar.deb: | ${ISTIO_OUT_LINUX} deb/fpm
+${ISTIO_OUT_LINUX}/release/istio-sidecar.rpm: | ${ISTIO_OUT_LINUX} rpm/fpm
+
+# Package the sidecar rpm file.
+rpm/fpm:
```

Comment:
Are these targets duplicated with istio.mk?

Reply:
I just moved them from there, that file is removed now

Reply:
Actually looks like I forgot to delete the other one

Reply:
Fixed